lib/compilers/pony.ts

```diff
@@ -0,0 +1,78 @@
+import path from 'path';
+
+import {ParseFilters} from '../../types/features/filters.interfaces';
+import {BaseCompiler} from '../base-compiler';
+import {CompilationResult, ExecutionOptions} from '../../types/compilation/compilation.interfaces';
+import _ from 'underscore';
+
+export class PonyCompiler extends BaseCompiler {
+    static get key() {
+        return 'pony';
+    }
+
+    /* constructor(info: any, env: any) {
+        super(info, env);
+
+        this.compiler.supportsIrView = true;
+        this.compiler.irArg = ['--pass', 'ir'];
+    } */
+
+    override optionsForFilter(filters: ParseFilters, outputFilename: any, userOptions?: any): string[] {
+        let options = ['-d', '-b', path.parse(outputFilename).name];
+
+        if (!filters.binary) {
+            options = options.concat(['--pass', 'asm']);
+        }
+
+        return options;
+    }
+
+    override preProcess(source: string, filters: any) {
+        // I do not think you can make a Pony "library", so you must always have a main.
+        // Looking at the stdlib, the main is used as a test harness.
+        if (!this.stubRe.test(source)) {
+            source += `\n${this.stubText}\n`;
+        }
+        return source;
+    }
+
+    override async generateIR(inputFilename: any, options: any[], filters: ParseFilters) {
```

Comment:
This is our fault because BaseCompiler still needs to get properly typed, but meanwhile the proper param types are:

```suggestion
    override async generateIR(inputFilename: string, options: string[], filters: ParseFilters) {
```